### PR TITLE
Support for streamline filter.

### PR DESCRIPTION
**Pull request by dpugmire**

### CODE REVIEW

File: src/vtkh/utils/Mutex.hpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#ifndef VTK_H_MUTEX_HPP
+#define VTK_H_MUTEX_HPP
+
+#ifdef ENABLE_OPENMP
```
Comment (mclarsen) [inline src/vtkh/utils/Mutex.hpp]: Does this work? I am not sure if this is defined when the source is compiling

**Comment by dpugmire** [comment]
oh crap,  it's VTKH_USE_OPENMP isn't it...

**Comment by dpugmire** [comment]
I've tested both with a different mechanism, so the code is good... I
changed the #ifdef when I was moving code around...

**Comment by mclarsen** [comment]
I don't think we are even setting that flag in the filters lib. I will get in the code and make sure flags get set.

**Comment by mclarsen** [comment]
I added a mutex cpp file so that the implementation and headers do not leak out.

File: src/vtkh/filters/CMakeLists.txt [hunk 2]
```diff
@@ -54,11 +75,16 @@ endif()
...
-  SOURCES ${vtkh_filters_sources}
-  HEADERS ${vtkh_filters_headers}
+  SOURCES ${vtkh_filters_sources} ${vtkh_comm_filters_sources}
+  HEADERS ${vtkh_filters_headers} ${vtkh_comm_filters_headers}
   DEPENDS_ON ${vtkh_filters_deps}
   )
 
+
+if(ENABLE_OPENMP)
+    blt_add_target_compile_flags(TO vtkh_filters FLAGS "-DVTKH_USE_OPENMP")
```
Comment (cyrush) [inline src/vtkh/filters/CMakeLists.txt:85]: does this require down stream folks to use this flag as well?
If not, seems ok.  If so, we should gen it in a header that folks include so they don't have to add defines.

File: src/vtkh/filters/CMakeLists.txt [hunk 3]
```diff
@@ -86,13 +124,21 @@ if (MPI_FOUND)
...
-      SOURCES ${vtkh_filters_sources}
-      HEADERS ${vtkh_filters_headers}
+      SOURCES ${vtkh_filters_sources} ${vtkh_comm_filters_sources} ${vtkh_mpi_sources}
+      HEADERS ${vtkh_filters_headers} ${vtkh_comm_filters_headers} ${vtkh_mpi_headers}
       DEPENDS_ON ${vtkh_filters_mpi_deps}
       )
+    if(ENABLE_OPENMP)
+          blt_add_target_compile_flags(TO vtkh_filters_mpi FLAGS "-DVTKH_USE_OPENMP")
+          if(CUDA_FOUND)
+              blt_add_target_compile_flags(TO vtkh_filters_mpi FLAGS "-Xcompiler ${OpenMP_CXX_FLAGS}")
```
Comment (cyrush) [inline src/vtkh/filters/CMakeLists.txt:136]: same q about the defines.


new blt might fix the XCompiler stuff, but should be safe

File: src/vtkh/utils/Logger.hpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  void WriteLog();
+protected:
+  DataLogger();
+  DataLogger(DataLogger const &);
+  std::stringstream Stream;
+  static class DataLogger* Instance;
+  std::stack<std::string> Entries;
+};
+
+#ifdef ENABLE_LOGGING
```
Comment (cyrush) [inline src/vtkh/utils/Logger.hpp:52]: should this become VTKH_ENABLE_LOGGING?

**Review by cyrush — APPROVED** [review]
Looks good, I had a few minor questions about defines.

**Pull request merged**